frontend-react/src/components/Content/MarkdownRenderer.tsx

```diff
@@ -26,6 +28,29 @@ type MarkdownContentProps = {
     markdownUrl: string;
 };
 
+type ReactMarkdownComponentsProp = Exclude<Options["components"], undefined>;
+export type ReactMarkdownComponentProps<
+    T extends string & keyof ReactMarkdownComponentsProp
+> = Extract<ReactMarkdownComponentsProp[T], (...args: any) => any> extends never
+    ? never
+    : Parameters<
+          Extract<ReactMarkdownComponentsProp[T], (...args: any) => any>
+      >[0];
+
+// Matches relative or cdc.gov absolute urls
+const INTERNAL_LINK_REGEX = /^(\/\w*?|(https:\/\/)?\w*?\.cdc\.gov)\/?.*$/;
```

Comment:
I know this is just a regular expression, but I think we could export it and add a few unit tests to make sure it catches what we need/doesn't catch what we don't want just to be safe -- would be especially useful if we need to update it in the future 🤷

Reply:
There's overlap here with work I just did with USLink in https://github.com/CDCgov/prime-reportstream/pull/8254 I feel. Should this PR pause until that completes so that the MarkdownRendererLink can do a is-route-or-cdcgov check, piggybacking off of regex introduced there?

Reply:
Sure, no rush and that makes sense to me!  Regular expressions just can be so finicky so it'd be great to add some small checks along the way :)